pyfr/plugins/tavg.py

```diff
@@ -351,3 +357,200 @@ def __call__(self, intg):
+            cfgsect = stat.get('tavg', 'cfg-section')
+            std = cfg.get(cfgsect, 'std-mode')
+            twindow = stat.getliteral('tavg', 'range')
+
+            for w in twindow:
+                times.append((Interval(*w), files, std))
+
+        active = times
+        for i, (r, f, s) in enumerate(times):
+            if any([(r in R and f != F) for R, F, S in active]):
+                active.pop(i)
+            elif any([r.o_intersection(R) for R, F, S in active]):
+                raise ValueError('Clash of average file time windows')
+        return active
+
+    def _compatibility_merge(self, active):
+        for t_rng, (file, stat, cfg), std_mode in active:
+            cfgsect = stat.get('tavg', 'cfg-section')
+            fields = stat.get('data', 'fields').split(',')
+
+            if self.uuid != file['mesh_uuid']:
+                raise RuntimeError('Average files computed on different meshes')
+            if self.region != cfg.get(cfgsect, 'region'):
+                raise RuntimeError('Average files computed on different regions')
+            if self.fields != fields:
+                raise RuntimeError('Average files computed on different fields')
+            if not all(cfg.get(cfgsect, k) == self.cfg.get(self.cfgsect, k)
+                       for k in cfg.items(cfgsect, prefix='avg-')):
+                raise RuntimeError('Different average field definitions')
+
+    def _eval_fun_exprs(self, avars):
+
+        # Prepare the substitution dictionary
+        subs = dict(zip(self.anames, avars))
+
+        # Evaluate the function and return
+        return np.array([npeval(v, subs) for v in self.fexprs])
+
+    def _init_tavg_merge(self, files):
+        file, stat, cfg = files
+        self.stat, self.cfg = stat, cfg
+        self.cfgsect = cfgsect = stat.get('tavg', 'cfg-section')
+
+        self.uuid = file['mesh_uuid']
+
+        self.fields = fields = stat.get('data', 'fields').split(',')
+        self.region = cfg.get(cfgsect, 'region')
+
+        self.dkeys = [k for k in file if k.startswith('tavg_')]
+
+        # Truth table to extract collumns later
+        self.aidx = aidx = [f.startswith('avg-') for f in fields]
+        self.sidx = sidx = [f.startswith('std-') for f in fields]
+        self.anames = [f.removeprefix('avg-') for f in compress(fields, aidx)]
+        self.snames = [f.removeprefix('std-') for f in compress(fields, sidx)]
+        nsnames = len(self.snames)
+
+        # Build fun avg expressions
+        c = cfg.items_as('constants', float)
+        self.fexprs = fexprs = []
+        for k in cfg.items(cfgsect, prefix='fun-avg-'):
+            fexprs.append(cfg.getexpr(cfgsect, k, subs=c))
+
+        # Initialise std max and avg registers
+        if self.std_all:
+            self.std_max = np.zeros(nsnames)
+            self.std_avg = np.zeros(nsnames)
+
+    def _merge_data(self, out, active, avg_time):
+        avg_time_inv = 1/avg_time
+        file0, _, _ = active[0][1]
```

Comment:
This is clunky, try and unpack active as just extra regular arguments to remove the first level of indexing.